dev/ProjectReunion_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddBootstrap.h"
+
+#include "IDynamicDependencyLifetimeManager.h"
+
+STDAPI AddFrameworkToPath(PCWSTR  path);
+
+IDynamicDependencyLifetimeManager* g_lifetimeManager = nullptr;
+DLL_DIRECTORY_COOKIE g_dllDirectoryCookie = 0;
+wil::unique_cotaskmem_string g_frameworkPath;
+
+STDAPI MddBootstrapInitialize(
+    const CLSID& appDynamicDependencyLifetimeManager)
+{
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager> lifetimeManager;
+    RETURN_IF_FAILED(CoCreateInstance(appDynamicDependencyLifetimeManager, nullptr, CLSCTX_LOCAL_SERVER, __uuidof(IDynamicDependencyLifetimeManager), reinterpret_cast<void**>(lifetimeManager.addressof())));
+
+    wil::unique_cotaskmem_string path;
+    RETURN_IF_FAILED(lifetimeManager->GetFrameworkPath(path.addressof()));
+
+    RETURN_IF_FAILED(AddFrameworkToPath(path.get()));
+
+    g_lifetimeManager = lifetimeManager.detach();
+    return S_OK;
+}
+
+STDAPI MddBootstrapShutdown()
+{
+    if (!g_lifetimeManager)
+    {
+        RETURN_IF_FAILED(g_lifetimeManager->Shutdown());
```

Comment:
Suggest `LOG_IF_FAILED()` here so the remainder of this method is nofail. (Close, release, free, and shutdown methods should generally return `void` as developers don't really have a recourse if it doesn't work.)